src/main/java/com/exasol/releasedroid/main/ReleaseDroid.java

```diff
@@ -24,38 +32,96 @@
     private final ReleaseUseCase releaseUseCase;
     private final ValidateUseCase validateUseCase;
     private final SummaryWriter summaryWriter;
+    private final RepositoryGateway repositoryGateway;
 
-    private ReleaseDroid(final ValidateUseCase validateUseCase, final ReleaseUseCase releaseUseCase) {
-        this.releaseUseCase = releaseUseCase;
+    /**
+     * Create a new instance of {@link ReleaseDroid}.
+     *
+     * @param repositoryGateway repository gateway
+     * @param releaseUseCase    release usecase
+     * @param validateUseCase   validate usecase
+     */
+    public ReleaseDroid(final RepositoryGateway repositoryGateway, final ValidateUseCase validateUseCase,
+            final ReleaseUseCase releaseUseCase) {
+        this.repositoryGateway = repositoryGateway;
         this.validateUseCase = validateUseCase;
+        this.releaseUseCase = releaseUseCase;
         this.summaryWriter = new SummaryWriter(new SummaryFormatter(new ReportFormatter()));
     }
 
-    public static ReleaseDroid of(final ValidateUseCase validateUseCase) {
-        return new ReleaseDroid(validateUseCase, null);
-    }
-
-    public static ReleaseDroid of(final ValidateUseCase validateUseCase, final ReleaseUseCase releaseUseCase) {
-        return new ReleaseDroid(validateUseCase, releaseUseCase);
-    }
-
     /**
      * Main entry point for all Release Droid's calls.
      */
     // [impl->dsn~rd-creates-validation-report~1]
     // [impl->dsn~rd-creates-release-report~1]
     public void run(final UserInput userInput) {
+        final Repository repository = this.repositoryGateway.getRepository(userInput);
+        validateUserInput(userInput, repository);
```

Comment:
I find it a bit confusing that this method is not only validating, but it's also changing the `UserInput`.

Maybe we can separate this "command" part to another method specific for it?

like `List<PlatformName> platformNames = getPlatformNames(userInput, repository.getPlatformNames());`

or something similar, and then pass that value to the use cases?